server/controllers/__tests__/datesControllers.test.js

```diff
@@ -0,0 +1,87 @@
+const controller = require('../datesController')
+
+const parseIt = require('../../services/Parser')
+const dateFixtures = require('../__fixtures__/dates')
+
+jest.mock('../../services/Parser')
+
+
+const res = {
+    json: jest.fn()
+}
+const log = jest.spyOn(global.console, "log").mockImplementation(() => {})
+
+beforeEach(() => {
+    jest.resetAllMocks()
+})
+
+describe('date controller test suite', () => {
+    const getDateMock = jest.fn()
+    const getAllDatesMock = jest.fn()
+    it('should return all dates if no date is provided', () => {
+        const req = {query: {}}
+        parseIt.mockImplementation(() => {
+            return {
+                getDate: getDateMock,
+                getAllDates: getAllDatesMock.mockReturnValue(dateFixtures)
+            }
+        })
```

Comment:
Not sure if we discussed this but this could get extracted using the method described at the top of the docs here. Not vital for this but it is often used. 
https://jestjs.io/docs/manual-mocks